main.py

```diff
@@ -1,16 +1,38 @@
-# This is a sample Python script.
+# 1.Напишите функцию, которая принимает на вход строку — абсолютный путь до файла.
+# Функция возвращает кортеж из трёх элементов: путь, имя файла, расширение файла.
+import os
 
-# Press Shift+F10 to execute it or replace it with your code.
-# Press Double Shift to search everywhere for classes, files, tool windows, actions, and settings.
 
+def split_path(file_path):
+    path, filename = os.path.split(file_path)
+    name, extension = os.path.splitext(filename)
+    return path, name, extension
 
-def print_hi(name):
-    # Use a breakpoint in the code line below to debug your script.
-    print(f'Hi, {name}')  # Press Ctrl+F8 to toggle the breakpoint.
 
+result_path = os.path.abspath('main.py')
+print(split_path(result_path))
 
-# Press the green button in the gutter to run the script.
-if __name__ == '__main__':
-    print_hi('PyCharm')
+# 2.Напишите однострочный генератор словаря, который принимает на вход три списка одинаковой длины:
+# имена str, ставка int, премия str с указанием процентов вида «10.25%». В результате получаем словарь с именем в качестве ключа и суммой премии в качестве значения.
+# Сумма рассчитывается как ставка умноженная на процент премии
 
-# See PyCharm help at https://www.jetbrains.com/help/pycharm/
+names = ["Иванов", "Петров", "Сидоров"]
+rates = [100, 200, 300]
+bonuses = ["10.25%", "15%", "5.5%"]
+
+result_gen_dict = {name: rate * float(bonus.strip("%")) / 100 for name, rate, bonus in zip(names, rates, bonuses)}
```

Comment:
Абсолютно верно, но избавиться от символа процента можно проще
```suggestion
result_gen_dict = {name: rate * float(bonus[:-1]) / 100 for name, rate, bonus in zip(names, rates, bonuses)}
```